gitcache/main.go

```diff
@@ -1,23 +1,62 @@
+// Copyright 2020 Security Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
 package main
 
 import (
+	"encoding/json"
 	"fmt"
-	"io/ioutil"
 	"log"
+	"net/http"
 	"os"
-	"path"
-	"time"
 
-	"github.com/go-git/go-git/v5"
-	"github.com/mholt/archiver"
-	"github.com/pkg/errors"
+	"github.com/ossf/scorecard/gitcache/pkg"
 	"go.uber.org/zap"
 )
 
-func main() {
-	const gittar string = "gitfolder.tar.gz"
-	var gitRepo *git.Repository
+type cache struct {
+	URL string `json:"url"`
+}
+
+var (
+	blob string
+	logf func(s string, f ...interface{})
+)
+
+func handler(w http.ResponseWriter, r *http.Request) {
+	switch r.Method {
+	case "POST":
+		d := json.NewDecoder(r.Body)
+		c := &cache{}
+		if err := d.Decode(c); err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest)
```

Comment:
should we return after all these Error calls?

Reply:
Friendly ping on this comment :) It doesn't really make sense to continue after these errors right?

Reply:
Sorry I didn't respond.You don't need to https://stackoverflow.com/a/40096885/19407

Reply:
Right, so subsequent error calls won't do anything, but we'll still end up unnecessarily calling `pkg.NewCacheService` and `UpdateCache` which may end up in undefined behaviour?

Reply:
even that assumption you can call http.Error multiple times may be wrong per the comment on https://golang.org/pkg/net/http/?m=all#Error: 

> Error replies to the request with the specified error message and HTTP code. **It does not otherwise end the request; the caller should ensure no further writes are done to w.** The error message should be plain text.

Reply:
I don't see that behavior in my test. In my next PR, I can do a return. I found another issue which I am going to fix soon.

Reply:
#240